src/misceval.js

```diff
@@ -1189,6 +1189,14 @@ Sk.misceval.applyOrSuspend = function (func, kwdict, varargseq, kws, args) {
         //append kw args to args, filling in the default value where none is provided.
         return func.apply(null, args);
     } else {
+        fcall = func.__get__;
+        if (fcall !== undefined) {
+            fcall = Sk.misceval.callsimOrSuspend(func.__get__, func, func);
+            //args.unshift(func);
+            if (fcall.tp$call) {
```

Comment:
Well the problem we're having here is that `func` itself is a suspension, so it will throw an exception even before it gets there. this call was just a shot in the dark. Thats why I asked for your help 😄.